1_fetch.smk

```diff
@@ -12,23 +12,29 @@ def get_mntoha_input_files(wildcards):
 
     '''
     out_dir = "1_fetch/out"
-    categories = config["mntoha"].keys()
+    categories = config["fetch_mntoha"].keys()
     input_files = []
     # include files under files key in each category
     for category in categories:
-        for filename in config["mntoha"][category]["files"]:
+        for filename in config["fetch_mntoha"][category]["files"]:
             input_files.append(os.path.join(out_dir, category + "_mntoha", filename))
     # include driver files: 3 types x 13 suffixes = 39 files
-    for driver_type in config["mntoha"]["drivers"]["driver_types"]:
-        for suffix in config["mntoha"]["drivers"]["suffixes"]:
+    for driver_type in config["fetch_mntoha"]["drivers"]["driver_types"]:
+        for suffix in config["fetch_mntoha"]["drivers"]["suffixes"]:
             input_files.append(os.path.join(out_dir, "drivers_mntoha", f"{driver_type}_{suffix}.zip"))
     return input_files
 
 
-rule fetch_all:
+checkpoint fetch_all:
     input:
         "1_fetch/in/pull_date.txt",
         get_mntoha_input_files
+    output:
+        "1_fetch/out/downloaded_files.txt"
+    run:
+        with open(output[0], 'w') as f:
+            for input_file in input[1]:
+                f.write(input_file + '\n')
```

Comment:
`fetch_all` is gone. It took some thinking to make this work because there's dynamic branching at play. In the old way I was doing it, fetch_all made sure EVERYTHING I knew I'd was was downloaded. It was a checkpoint because I needed to use `lake_metadata.csv` to determine outputs of downstream rules (dynamic branching). In the new way, I have a new checkpoint, `fetch_mntoha_metadata`, which downloads `lake_metadata.csv` only. Later rules that depend on `lake_metadata.csv` for dynamic branching call this checkpoint in their input functions.

The other big change I needed to make had to do with how to unzip files. Lake sequences need unzipped driver, clarity, and ice flags files. The unzipped files should depend on the unzipping rule in the DAG. Unzipping can be a one-to-many operation where the number of outputs varies - dynamic branching! So, I made another checkpoint, `unzip_archive`. This does the work of unzipping, and its output is the directory that files are unzipped to. Then, any rule that relies on unzipped files calls the `unzip_archive` checkpoint in its input function.